tests/js/spec/views/organizationEventsV2/utils.spec.jsx

```diff
@@ -52,7 +53,65 @@ describe('getQuery()', function() {
     };
 
     expect(getQuery(view, {}).query).toEqual('event.type:csp');
-    expect(getQuery(view, {query: {query: 'test'}}).query).toEqual('test event.type:csp');
+  });
+
+  it('appends query conditions in location', function() {
+    const view = {
+      id: 'test',
+      name: 'test view',
+      data: {fields: ['id'], query: 'event.type:csp'},
+      tags: [],
+    };
+    const location = {
+      query: {
+        query: 'TypeError',
+      },
+    };
+    expect(getQuery(view, location).query).toEqual('event.type:csp TypeError');
+  });
+});
+
+describe('getQueryString()', function() {
+  it('excludes empty values', function() {
+    const view = {
+      data: {
+        query: 'event.type:transaction',
+      },
+    };
+    const location = {
+      query: {query: ''},
+    };
+    expect(getQueryString(view, location)).toEqual('event.type:transaction');
+  });
+
+  it('includes view, and location query data', function() {
+    const view = {
+      data: {
+        query: 'event.type:transaction',
+      },
+    };
+    const location = {
+      query: {query: 'TypeError'},
+    };
+    expect(getQueryString(view, location)).toEqual('event.type:transaction TypeError');
+  });
+
+  it('includes non-empty additional data', function() {
+    const view = {
+      data: {
+        query: 'event.type:transaction',
+      },
+    };
+    const location = {};
+    const additional = {
+      nope: '',
+      undef: undefined,
```

Comment:
Are we concerned about ints? e.g. 0

Reply:
I'm not right now. The main use-case I have for the additional parameters are to factor in issue ids and transaction names, which are both string values.